bolides/bdf.py

```diff
@@ -300,10 +388,103 @@ def get_closest_by_loc(self, lon, lat, n=1):
             The filtered data.
 
         """
-        lon_diff = (self['longitude'] - lon).abs()
-        lat_diff = (self['latitude'] - lat).abs()
-        tot_diff = lon_diff + lat_diff
-        return self.iloc[tot_diff.argsort()].head(n)
+        distances = haversine(lat, lon, self['latitude'].values, self['longitude'].values)
+        return self.iloc[distances.argsort()].head(n)
+
+    def get_closest(self, datestr=None, lon=None, lat=None, k=1, time_weight=1, loc_weight=1, time_scale=86400, loc_scale=100):
+        """
+        Get the n bolides closest to a given iso-format date string and to a given longitude and latitude in degrees. This is calculated using BallTree and a custom distance metric that finds the expected number of pairs of bolides that would be at least as close in space and time as the two points being compared, under a random (uniform) distribution. Effectively, given a time and location, this will find and cluster bolide events that are nearest to it.
```

Comment:
Are you familiar with PEP 8 for Python coding standards? I'm not religious about following PEP 8 but I try to be consistent with it. Your comment line here is 467 characters long. That will wrap or go off screen for all but the widest of monitors! PEP 8 says no more than 79 characters per line. I tend to relax that a bit and go up to 110 characters. Please format your text lines to be no more than 110 characters long. Same with other liens in your code below.